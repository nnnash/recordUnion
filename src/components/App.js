import React, {Fragment} from 'react';

import {withDelayAppear} from './hocs';
import NotesList from './NotesList';
import Dialog from './Dialog';
import './style.css';

const DelayedNotesList = withDelayAppear(NotesList);
const DelayedDialog = withDelayAppear(Dialog);

export default () => (
    <Fragment>
        <DelayedNotesList delay={2000} noteDelay={2000} />
        <DelayedDialog delay={1000} />
    </Fragment>
);
