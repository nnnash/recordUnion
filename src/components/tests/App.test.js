import React from 'react';
import {mount} from 'enzyme';

import App from '../App';

jest.mock('../NotesList', () => {
    const NotesList = ({children}) => children || '';

    return NotesList;
});
jest.mock('../Dialog', () => {
    const Dialog = ({children}) => children || '';

    return Dialog;
});
jest.mock('../hocs', () => {
    const {mapProps} = require('recompose');
    const withDelayAppear = mapProps(({delay, ...rest}) => rest);

    return {
        withDelayAppear: withDelayAppear,
    };
});

describe('App.js', () => {
    it('should render both Dialog and NotesList', () => {
        expect(mount(<App />)).toMatchSnapshot();
    });
});
