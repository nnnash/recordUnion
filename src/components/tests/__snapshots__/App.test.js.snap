// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App.js should render both Dialog and NotesList 1`] = `
<_default>
  <mapProps(NotesList)
    delay={2000}
    noteDelay={2000}
  >
    <NotesList
      noteDelay={2000}
    />
  </mapProps(NotesList)>
  <mapProps(Dialog)
    delay={1000}
  >
    <Dialog />
  </mapProps(Dialog)>
</_default>
`;
